### PR TITLE
sensible-utils: init at 0.0.18

### PR DESCRIPTION
**Pull request by pbek**
###### Description of changes

Some packages, like [pet](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/tools/pet/default.nix) and [qc](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/tools/qc/default.nix) depend on "sensible-editor" to select an editor for editing its config file.

This package provides that script via Debian's [sensible-utils](https://salsa.debian.org/debian/sensible-utils).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pbek** [comment]
Result of `nixpkgs-review pr 237715` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sensibleUtils</li>
  </ul>
</details>

File: pkgs/tools/misc/sensible-utils/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    wrapProgram $out/bin/sensible-pager \
+      --prefix PATH : ${lib.makeBinPath [ bash ]}
+
+    cp sensible-terminal $out/bin/sensible-terminal
+    wrapProgram $out/bin/sensible-terminal \
+      --prefix PATH : ${lib.makeBinPath [ bash ]}
+  '';
+
+  meta = with lib; {
+    description = "This package provides a number of small utilities which are used by programs to sensibly select and spawn an appropriate browser, editor, or pager. The specific utilities included are: sensible-browser sensible-editor sensible-pager";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sensible-utils/default.nix:44]:
```suggestion
    description = "The package provides utilities used by programs to sensibly select and spawn an appropriate browser, editor, or pager.";
    longDescription = ''
       The specific utilities included are:
       - sensible-browser
       - sensible-editor
       - sensible-pager
    '';
```
that was a bit long for description, I tried to shorten it a bit.

File: pkgs/tools/misc/sensible-utils/default.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{ stdenv, lib, fetchgit, makeWrapper, bash }:
+
+let
+  pname = "sensible-utils";
+  version = "0.0.18";
+in
+stdenv.mkDerivation rec {
+  inherit pname version;
+
+  # https://salsa.debian.org/debian/sensible-utils
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sensible-utils/default.nix:10]:
```suggestion
```
when I open the git link, I land there

File: pkgs/tools/misc/sensible-utils/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{ stdenv, lib, fetchgit, makeWrapper, bash }:
+
+let
+  pname = "sensible-utils";
+  version = "0.0.18";
+in
+stdenv.mkDerivation rec {
+  inherit pname version;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sensible-utils/default.nix:8]:
```suggestion
stdenv.mkDerivation rec {
  pname = "sensible-utils";
  version = "0.0.18";
```

File: pkgs/tools/misc/sensible-utils/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    wrapProgram $out/bin/sensible-editor \
+      --prefix PATH : ${lib.makeBinPath [ bash ]}
+
+    cp sensible-pager $out/bin/sensible-pager
+    wrapProgram $out/bin/sensible-pager \
+      --prefix PATH : ${lib.makeBinPath [ bash ]}
+
+    cp sensible-terminal $out/bin/sensible-terminal
+    wrapProgram $out/bin/sensible-terminal \
+      --prefix PATH : ${lib.makeBinPath [ bash ]}
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sensible-utils/default.nix:40]:
```suggestion
    cp sensible-browser sensible-editor sensible-pager sensible-terminal $out/bin/
```
those don't exec to bash and the shebang gets fixed in the fixupPhase

File: pkgs/tools/misc/sensible-utils/default.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    url = "https://salsa.debian.org/debian/sensible-utils.git";
+    rev = "debian/${version}";
+    sha256 = "sha256-fZJKPnEkPfo/3luUcHzAmGB2k1nkA4ATEQMSz0aN0YY=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  configurePhase = ''
+    # We don't want to use configure!
+  '';
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sensible-utils/default.nix:21]:
```suggestion
  dontConfigure = true;
```

File: pkgs/tools/misc/sensible-utils/default.nix [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+let
+  pname = "sensible-utils";
+  version = "0.0.18";
+in
+stdenv.mkDerivation rec {
+  inherit pname version;
+
+  # https://salsa.debian.org/debian/sensible-utils
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sensible-utils/default.nix:11]: We can't use fetchFromGitLab?

File: pkgs/tools/misc/sensible-utils/default.nix (hunk 4 above)
Comment (pbek) [reply pkgs/tools/misc/sensible-utils/default.nix:40]: nice, thank you!

File: pkgs/tools/misc/sensible-utils/default.nix (hunk 5 above)
Comment (pbek) [reply pkgs/tools/misc/sensible-utils/default.nix:17]: Ah, there is an option for this! Thank you!

File: pkgs/tools/misc/sensible-utils/default.nix (hunk 6 above)
Comment (pbek) [reply pkgs/tools/misc/sensible-utils/default.nix:7]: Ah, you can enter the "domain". Thank you!

**Comment by pbek** [comment]
Thanks a lot for your feedback and changes, @SuperSandro2000! I integrated and squashed.

**Comment by SuperSandro2000** [comment]
@ofborg build sensibleUtils

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -1162,6 +1162,8 @@ with pkgs;
 
   closureInfo = callPackage ../build-support/closure-info.nix { };
 
+  sensibleUtils = callPackage ../tools/misc/sensible-utils { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:1165]:
```suggestion
  sensible-utils = callPackage ../tools/misc/sensible-utils { };
```
that makes more sense and also auto triggers ofborg.

**Pull request merged**